docs/source/roadmap.md

```diff
@@ -0,0 +1,266 @@
+
+### A Developer Focus Point
+- [ ] Encourage groups we meet to contribute development resources to _starfish_
+- [ ] Support image-based transcriptomics and proteomics developers who are interested in working with us on _starfish_ and have implemented methods that could synergize with _starfish_ by scoping how they could contribute, and supporting any eventual pull requests made to _starfish_.
+- [ ] Serve Data To Community
+- [ ] CC-by licensing
+- [ ] Documented upload procedures to s3
+
+## Candidate features for 0.2.0+
+
+There are many other features that are good candidates for future releases. We discuss these features, and why we believe they can be built after release 0.1.0.
+
+### Hardware- and Acquisition-related corrections
+Data generators understand their hardware, chemistry, and samples much better than we do. For now, it's difficult for the _starfish_ team to solve image pre-processing problems that are specific to the signal acquisition challenges each data generator faces. Furthermore, proper image pre-processing is critical to the success of the downstream pipeline components that we're building in the roadmap defined above. Instead of relying on data generators to provide us with image-pre-processing pipeline component implementations, we're requesting that data generators pre-process their images before uploading. This will accelerate progress towards the project's goal of comparing the results from all the methods using standard file formats, and scalable, reproducible image processing workflows. There are three categories of pre-processing that we ask data generators to apply:
+
+#### Affine Registration
+For the SpaceTx pilot project, it is adequate to apply pre-computed transformations or leverage already-registered data. Later, when we want to support groups to use _starfish_ for their own research, it will be important to implement affine registration, as otherwise _starfish_ serves only part of their use cases, and they are incentivized to continue running their existing pipelines.
+
+#### Non-Affine Registration
+Non-Affine registration approaches are very specific to the types of deformations that the tissue suffers, and have variable forms. We should investigate defining some kind of API for arbitrary transformations that could allow users to add approaches that they require, but we won't do the work to implement these as they don't have broad applicability.
+
+#### Fixing Chromatic Aberration
+Chromatic aberrations, including differences in illumination, cross-talk between channels, and channel drift tend to be specific to the assay and microscope, making creation of a general solution complicated. Additionally, these problems are often fixed early in pipelines so it is easy to extricate this step from the pipeline. For these reasons, we will ask the groups to generate data that has already been corrected for release 0.1.0.
```

Comment:
While it is true that these corrections can be more easily performed outside of starfish because they are early in the workflow, I think they really should be done in starfish so they are logged. These types of corrections can have a large effect on the final result (e.g., pixel wise decoding using a euclidian distance to assign targets assumes certain relative intensities between channels/rounds), so I think it is important that they are logged with the rest of the processing steps to improve reproducibility (between experiments, not re-running an analysis) and comparison between datasets.

Also, I think that (technically) cross-talk and a lot of the differences in measured intensity are not chromatic aberrations :P

Reply:
> Also, I think that (technically) cross-talk and a lot of the differences in measured intensity are not chromatic aberrations :P

What would you call them instead? 

> While it is true that these corrections can be more easily performed outside of starfish because they are early in the workflow, I think they really should be done in starfish so they are logged. These types of corrections can have a large effect on the final result (e.g., pixel wise decoding using a euclidian distance to assign targets assumes certain relative intensities between channels/rounds), so I think it is important that they are logged with the rest of the processing steps to improve reproducibility (between experiments, not re-running an analysis) and comparison between datasets.

Completely agree 👍. We intend to include some solution(s) for these problems (and therefore add them to provenance logging), but we don't have enough computational biology hours in the coming quarter, and we won't have time to add these capabilities prior to 0.1.0. 

That said, if you wanted to _contribute_ such solutions needed for _your system_, that would be very valuable and a model for future groups :-).

Reply:
>> Also, I think that (technically) cross-talk and a lot of the differences in measured intensity are not chromatic aberrations :P
> 
> What would you call them instead?

Chromatic aberration arises from changes in refractive index as a function of wavelength (i.e., dispersion). Cross-talk between different channels is due to the microscope filters not rejecting fluorescence from other dyes (may not be possible due to overlap in spectra). The difference in intensities across channels is usually due to differences in illumination intensity, properties of the dyes, and the sensitivity of the detector. All of that being said, I can't think of a catch all term at the moment :P

After re-reading, I am confused about what this section refers to. Is this section proposing ways to automatically determine the transformations required to make the corrections or is it proposing to apply corrections determined/defined the by user? If it's the latter, I think there is a lot of overlap with the "Image re-scaling & normalization" section.

Reply:
> That said, if you wanted to contribute such solutions needed for your system, that would be very valuable and a model for future groups :-).

I am definitely interested. We need to do some cross-talk correction in upcoming experiments, so I can work on at least that part soon.

Reply:
> Is this section proposing ways to automatically determine the transformations required to make the corrections or is it proposing to apply corrections determined/defined the by user? If it's the latter, I think there is a lot of overlap with the "Image re-scaling & normalization" section.

Like other stuff in this section, I have less experience with these image processing. It's "one or the other, depending on how hard they turn out to be". I think this is worth bringing up at the sync, as there's complexity here I'd like to unpack through conversation.

Reply:
Sounds good! I look forward to discussing at the sync.